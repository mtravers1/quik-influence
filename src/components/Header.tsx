import React from "react";
import {
  Box,
  Stack,
  Heading,
  Flex,
  Text,
  Button,
  useDisclosure,
  ColorModeScript,
  Image,
  HStack,
  Avatar,
  Tag,
  TagLabel
} from "@chakra-ui/react";
import { HamburgerIcon } from "@chakra-ui/icons";
import { theme } from "../../pages/_app";
import AppIcon from "../assets/icon.png";
import quikColorConstants from "../utils/colorConstants";

const Header = ({ ...rest }) => {
  const { isOpen, onOpen, onClose } = useDisclosure();
  const handleToggle = () => (isOpen ? onClose() : onOpen());

  return (
    <Flex
      as="nav"
      align="center"
      justify="space-between"
      wrap="wrap"
      padding={6}
      color="black"
      boxShadow="base"
      {...rest}
    >
      <Flex align="center" mr={5}>
        <Image
          boxSize="30px"
          objectFit="cover"
          src={AppIcon.src}
          alt="quik-influence logo"
        />
        <Heading as="h1" size="lg" ml={3} letterSpacing={"tighter"}>
          <Flex>
            <Text color={quikColorConstants.influenceRed} mr={1}>
              Quik
            </Text>
            <Text>Influence</Text>
          </Flex>
        </Heading>
      </Flex>

      <Box display={{ base: "block", md: "none" }} onClick={handleToggle}>
        <HamburgerIcon />
      </Box>

      {/* <Stack
        direction={{ base: "column", md: "row" }}
        display={{ base: isOpen ? "block" : "none", md: "flex" }}
        width={{ base: "full", md: "auto" }}
        alignItems="center"
        flexGrow={1}
        mt={{ base: 4, md: 0 }}
      >
        <Text>Docs</Text>
        <Text>Examples</Text>
        <Text>Blog</Text>
      </Stack> */}

      <Box>
        <ColorModeScript initialColorMode={theme.config.initialColorMode} />
      </Box>

      <Box
        display={{ base: isOpen ? "block" : "none", md: "block" }}
        mt={{ base: 4, md: 0 }}
      >
        <Button
          variant="outline"
          _hover={{ bg: "grey.700", borderColor: "teal.700" }}
        >
          Create account
        </Button>
        <Tag size="lg" colorScheme="grey.500" borderRadius="full">
          <Avatar
            src="https://bit.ly/sage-adebayo"
            size="md"
            name="Segun Adebayo"
            ml={1}
            mr={2}
          />
          <TagLabel>Segun</TagLabel>
        </Tag>
      </Box>
    </Flex>
  );
};

export default Header;
