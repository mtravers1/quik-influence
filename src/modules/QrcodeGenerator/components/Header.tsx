import { Flex, Heading } from '@chakra-ui/react';

import { QRIcon } from './Icons';
import QRSettings from './QRSettings';

export default function Header() {
  return (
    <Flex as="header" justify="space-between" align="center" shadow="md" p="5">
      <Heading fontSize={['xl', '2xl', '4xl']}>
        QRCode Generator <QRIcon ml="2" fontSize="larger" />
      </Heading>
      <QRSettings />
    </Flex>
  );
}
