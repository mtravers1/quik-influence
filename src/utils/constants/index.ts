export const CONTENT_URL = '/content';
export const Q_TOKEN = '_q_inf';
export const DEFAULT_PAGE_SIZE = '100';
export const ADMINS_ID = [2];

export const FILE_STATUS = {
  PENDING: 'pending',
  PROCESSING: 'processing',
  UPLOADING: 'uploading',
  PAUSED: 'paused',
  COMPLETED: 'completed',
  FAILED: 'failed',
  IDLE: 'idle',
};

export const headers: string[] = [
  'phone',
  'avatar',
  'email',
  'firstName',
  'lastName',
  'address',
  'city',
  'dateOfBirth',
  'instagramId',
  'socialMediaHandle',
  'state',
  'country',
  'postalCode',
  'gender',
  'address2',
  'address3',
  'facebookHandle',
  'tiktokHandle',
  'twitterHandle',
];

export const formDataRelatedToSpecificUser = [
  'updatedAt',
  'twitterHandle',
  'avatar',
  'phone',
  'address2',
  'email',
  'address1',
  'postalCode',
  'socialMediaHandle',
  'lastName',
  'firstName',
  'instagramId',
  'tiktokHandle',
  'address',
  'address3',
  'facebookHandle',
];

export const MARKETING_ADMIN = ['can_approve_or_reject_campaign'];
export const ADMIN_USER_AFFILIATE = ['can_get_joinable_campaigns'];
export const CAN_HANDLE_INFLUENCERS = ['can_manage_influencers'];

export const MARKET_PLACE_NEW_PRODUCTS = 'new';
export const MARKET_PLACE_POPULAR_PRODUCTS = 'popular';

export const CART_CLICK_NAME = 'cart';
