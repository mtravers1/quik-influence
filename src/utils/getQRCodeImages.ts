import QRCode, { QRCodeToDataURLOptions } from 'qrcode';
import { QrCodeRawData } from 'types';

const defaultOptions: Partial<QRCodeToDataURLOptions> = {
  errorCorrectionLevel: 'M',
  width: 256,
};

export default async function getQRImages(
  list: QrCodeRawData[],
  options?: QRCodeToDataURLOptions,
): Promise<string[]> {
  const qrCodeUrlPromises = list.map(({ text }) =>
    QRCode.toDataURL(text, { ...defaultOptions, ...options }),
  );
  const images = await Promise.all(qrCodeUrlPromises);
  return images;
}
