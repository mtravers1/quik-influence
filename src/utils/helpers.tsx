import axios from 'axios';

// const baseurl = 'http://localhost:2022';
const baseurl = process.env.NEXT_PUBLIC_BACKEND_URL;

import { DropdownSelectOption } from 'components/DropdownSelect';

const _axiosInstance = axios.create({
  baseURL: `${baseurl}/api/v1`,
  headers: {
    'Access-Control-Allow-Headers':
      'Access-Control-Allow-Origin, Access-Control-Allow-Headers, Content-Type',
    'Access-Control-Allow-Origin': '*',
  },
});

_axiosInstance.interceptors.request.use((config: any) => {
  let token: any = '';
  if (typeof window !== 'undefined') {
    token = localStorage.getItem('token');
  }
  config.headers.token = token;
  return config;
});

export const axiosInstance = _axiosInstance;

export const validate = (field: any, Regex: any, pattern: any) => {
  if (pattern.test(field)) return true;
  return false;
};

export const setToken = (token: string) => {
  axiosInstance.interceptors.request.use((config: any) => {
    config.headers.token = token ? token : '';
    return config;
  });
};

export const getNumberRange = (
  start: number,
  stop: number,
  step: number
): DropdownSelectOption[] =>
  Array.from({ length: (stop - start) / step + 1 }, (_, i) => ({
    label: (start + i * step).toString(),
    value: (start + i * step).toString(),
  }));
