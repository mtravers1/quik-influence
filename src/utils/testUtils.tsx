import { FC, ReactElement } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from 'redux/reducers';

function makeTestStore(rootReducer: any, initialState: any) {
  const store = createStore(rootReducer, initialState, applyMiddleware(thunk));
  const origDispatch = store.dispatch;
  store.dispatch = jest.fn(origDispatch);
  return store;
}

const renderWithStore = (ui: ReactElement, storedefaults: any) => {
  const { initialState, ...renderOptions } = storedefaults;
  const store = makeTestStore(rootReducer, initialState);

  const Wrapper: FC = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export * from '@testing-library/react';
export { renderWithStore };
