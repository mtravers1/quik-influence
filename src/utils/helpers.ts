import axios from 'axios';
import { omitBy, isNil } from 'lodash';
import { ADMINS_ID, Q_TOKEN } from './constants';

import { DropdownSelectOption } from 'components/DropdownSelect';
import { FilterDataProps } from 'types';
import { format } from 'date-fns';

export const baseurl = process.env.BACKEND_URL;

export const axiosInstance = axios.create({
  baseURL: `${baseurl}/api/v1`,
  // withCredentials: true,
  headers: {
    'Access-Control-Allow-Headers':
      'Access-Control-Allow-Origin, Access-Control-Allow-Headers, Content-Type',
    'Access-Control-Allow-Origin': '*',
  },
});

export const logout = () => {
  localStorage.removeItem(Q_TOKEN);
  window.location.href = '/login';
};

export const validate = (field: any, pattern: any) => {
  const parts = /\/(.*)\/(.*)/.exec(pattern) || [];
  let restoredRegex = new RegExp(parts[1], parts[2]);
  if (restoredRegex.test(field)) return true;
  return false;
};

export const setToken = (token: string) => {
  axiosInstance.defaults.headers.common['token'] = token;

  if (typeof window !== 'undefined') {
    localStorage.setItem(Q_TOKEN, token);
  }
};

const tokens: any = {};

export function parseJwt(token: any) {
  if (!token) return;
  if (tokens[token]) return tokens[token];

  const base64Url = token.split('.')[1];
  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
  const jsonPayload = decodeURIComponent(
    atob(base64)
      // Buffer.from(base64, 'base64')
      .split('')
      .map(c => {
        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
      })
      .join('')
  );

  const result = JSON.parse(jsonPayload);
  tokens[token] = result;

  return result;
}

export function getUser() {
  let user;
  let ctoken;
  let isExpired: boolean = false;

  if (typeof window !== 'undefined') {
    ctoken = localStorage.getItem(Q_TOKEN);
  }

  if (ctoken !== 'null') {
    user = parseJwt(ctoken);
  }

  isExpired = user && user.exp && user.exp < Date.now() / 1000;

  return { admin: user, token: ctoken, isExpired };
}

export const getNumberRange = (
  start: number,
  stop: number,
  step: number
): DropdownSelectOption[] =>
  Array.from({ length: (stop - start) / step + 1 }, (_, i) => ({
    label: (start + i * step).toString(),
    value: (start + i * step).toString(),
  }));

export const getQueryString = (params?: FilterDataProps) => {
  const paramsFilters = omitBy(params, isNil);
  const query = Object.keys(paramsFilters)
    .map(
      k => encodeURIComponent(k) + '=' + encodeURIComponent(paramsFilters[k])
    )
    .join('&');
  return query;
};

export const getSocialHandleHeader = (socialColumns: string[]): string[] => {
  let socialHeader: string[] = [];
  socialColumns.forEach((socialColumn: any) => {
    switch (socialColumn) {
      case 'facebookHandle':
        socialHeader.push('Facebook');
        break;
      case 'twitterHandle':
        socialHeader.push('Twitter');
        break;
      case 'instagramId':
        socialHeader.push('Instagram');
        break;
      case 'tiktokHandle':
        socialHeader.push('Tik Tok');
        break;
      default:
        break;
    }
  });
  return socialHeader;
};

export const similarity = (s1: string, s2: string) => {
  let longer = s1;
  let shorter = s2;
  if (s1.length < s2.length) {
    longer = s2;
    shorter = s1;
  }
  let longerLength = longer.length;
  if (longerLength == 0) {
    return 1.0;
  }
  return (longerLength - editDistance(longer, shorter)) / longerLength;
};

function editDistance(s1: string, s2: string) {
  s1 = s1.toLowerCase();
  s2 = s2.toLowerCase();

  let costs = new Array();
  for (let i = 0; i <= s1.length; i++) {
    let lastValue = i;
    for (let j = 0; j <= s2.length; j++) {
      if (i == 0) costs[j] = j;
      else {
        if (j > 0) {
          let newValue = costs[j - 1];
          if (s1.charAt(i - 1) != s2.charAt(j - 1))
            newValue = Math.min(Math.min(newValue, lastValue), costs[j]) + 1;
          costs[j - 1] = lastValue;
          lastValue = newValue;
        }
      }
    }
    if (i > 0) costs[s2.length] = lastValue;
  }
  return costs[s2.length];
}

export const stringSearch = (val: string, string: string) =>
  string && string.toLowerCase().search(val.toLowerCase()) !== -1;

export function isInViewport(element: any) {
  const rect = element.getBoundingClientRect();
  return (
    rect.top >= 0 &&
    rect.left >= 0 &&
    rect.bottom <=
      (window.innerHeight || document.documentElement.clientHeight) &&
    rect.right <= (window.innerWidth || document.documentElement.clientWidth)
  );
}

export const isAdmin = () => {
  const user = getUser();
  return user.admin && ADMINS_ID.includes(user.admin.roleId);
};

export const hasPermission = (
  userPerm: string[] | undefined,
  permission: string[] | undefined
) => {
  if (userPerm?.length === 0) return true;
  return userPerm?.some(perm => permission?.includes(perm));
};

export const getReportOpenDate = (events: any) => {
  if (!events) return 'NA';
  const openEvent = events.find((evt: any) => evt['event_name'] === 'open');
  if (!openEvent) return 'Not opened yet';

  return format(new Date(openEvent['processed']), 'dd/MM/yyyy');
};

export const getReportUnsub = (events: any) => {
  if (!events) return 'NA';
  const unsubEvent = events.find(
    (evt: any) => evt['event_name'] === 'unsubscribe'
  );
  if (!unsubEvent) return 'No';

  return 'Yes';
};

export const truncateText = (str: string, num: number) => {
  if (str.length <= num) {
    return str;
  }
  return str.slice(0, num) + '...';
};
