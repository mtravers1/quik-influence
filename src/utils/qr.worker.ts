import { QRCodeToDataURLOptions } from 'qrcode';
import { QrCodeRawData } from 'types';

import getQRCodeImages from './getQRCodeImages';

export async function generateImages(
  values: QrCodeRawData[],
  options?: QRCodeToDataURLOptions,
): Promise<string[]> {
  const qrCodeImages = await getQRCodeImages(values, options);
  return qrCodeImages;
}
