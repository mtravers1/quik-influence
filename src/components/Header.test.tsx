import { renderWithStore } from 'utils/testUtils';
import initialState from '__mockData__/storeData';
import Header from './Header';

it('renders Header component', () => {
  const { container } = renderWithStore(<Header />, { initialState });
  expect(container).toMatchSnapshot();
});
