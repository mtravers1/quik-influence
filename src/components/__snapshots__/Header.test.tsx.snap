// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header component 1`] = `
<div>
  <nav
    class="css-mm8nhs"
  >
    <div
      class="css-1c714fi"
    >
      <img
        alt="quik-influence logo"
        class="chakra-image css-1ko1n4n"
      />
      <h1
        class="chakra-heading css-8mmjuy"
      >
        <div
          class="css-k008qs"
        >
          <p
            class="chakra-text css-67ii6d"
          >
            Quik
          </p>
          <p
            class="chakra-text css-0"
          >
            Influence
          </p>
        </div>
      </h1>
    </div>
    <div
      class="css-dsyim5"
    >
      <svg
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="css-0"
    >
      <script>
        (function setScript(initialValue) {
  var mql = window.matchMedia("(prefers-color-scheme: dark)");
  var systemPreference = mql.matches ? "dark" : "light";
  var persistedPreference = systemPreference;

  try {
    persistedPreference = localStorage.getItem("chakra-ui-color-mode");
  } catch (error) {
    console.log("Chakra UI: localStorage is not available. Color mode persistence might not work as expected");
  }

  var colorMode;

  if (persistedPreference) {
    colorMode = persistedPreference;
  } else if (initialValue === "system") {
    colorMode = systemPreference;
  } else {
    colorMode = initialValue != null ? initialValue : systemPreference;
  }

  if (colorMode) {
    /**
     * Keep in sync with \`root.set() {@file ./color-mode.utils.ts}
     */
    document.documentElement.style.setProperty("--chakra-ui-color-mode", colorMode);
    document.documentElement.setAttribute("data-theme", colorMode);
  }
})('dark')
      </script>
    </div>
    <div
      class="css-iqqio3"
    >
      <button
        aria-controls="menu-list-6"
        aria-expanded="false"
        aria-haspopup="menu"
        class="chakra-menu__menu-button css-1eqi7qs"
        id="menu-button-6"
      >
        <span
          class="css-xl71ch"
        >
          <span
            class="css-1ehzxp"
          >
            <span
              class="chakra-avatar css-o11zrc"
            >
              <div
                aria-label=" "
                class="chakra-avatar__initials css-0"
                role="img"
              />
            </span>
            <span
              class="css-8uhtka"
            >
               
              
            </span>
          </span>
        </span>
      </button>
      <div
        class="css-0"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-sjt5nk"
          id="menu-list-6"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <button
            class="chakra-menu__menuitem css-1k756t9"
            data-index="0"
            id="menu-list-6-menuitem-1"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Theme: 
            <label
              class="chakra-switch css-qtgvng"
            >
              <input
                class="chakra-switch__input"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-switch__track css-14qxnv8"
              >
                <span
                  class="chakra-switch__thumb css-0"
                />
              </span>
            </label>
          </button>
          <button
            class="chakra-menu__menuitem css-1k756t9"
            data-index="1"
            id="menu-list-6-menuitem-2"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Profile
          </button>
           
          <button
            class="chakra-menu__menuitem css-1k756t9"
            data-index="2"
            id="menu-list-6-menuitem-3"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Edit profile
          </button>
          <button
            class="chakra-menu__menuitem css-1k756t9"
            data-index="3"
            id="menu-list-6-menuitem-4"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Log out
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
